frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx

```diff
@@ -242,6 +242,66 @@ describe("StringFilterValuePicker", () => {
       expect(onChange).toHaveBeenCalledWith(["t", "p"]);
     });
 
+    it("should elevate selected field values on initial render", async () => {
+      await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["p"],
+        fieldValues: createMockFieldValues({
+          field_id: PRODUCTS.CATEGORY,
+          values: [
+            ["t", "To-do"],
+            ["p", "In-progress"],
+            ["c", "Completed"],
+          ],
+        }),
+      });
+
+      const checkboxes = screen.getAllByRole("checkbox");
+      expect(checkboxes[0]).toHaveAccessibleName("In-progress");
+      expect(checkboxes[0]).toBeChecked();
+      expect(checkboxes[1]).toHaveAccessibleName("To-do");
+      expect(checkboxes[1]).not.toBeChecked();
+      expect(checkboxes[2]).toHaveAccessibleName("Completed");
+      expect(checkboxes[2]).not.toBeChecked();
+    });
+
+    it("should not elevate selected field values after the initial render", async () => {
+      const { rerender, onChange } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["p"],
+        fieldValues: createMockFieldValues({
+          field_id: PRODUCTS.CATEGORY,
+          values: [
+            ["t", "To-do"],
+            ["p", "In-progress"],
+            ["c", "Completed"],
+          ],
+        }),
+      });
+
+      rerender(
+        <StringFilterValuePicker
+          query={query}
+          stageIndex={stageIndex}
+          column={column}
+          values={["p", "c"]}
```

Comment:
Let's also make sure that unchecking values doesn't move the items.
This can be a separate test case, or this one:

```suggestion
          values={["c"]}
```

plus `expect(checkboxes[0]).not.toBeChecked();`

Reply:
Added a test